### PR TITLE
Posenet - allow width and height specification for input resolution & relax input resolution requirement

### PR DESCRIPTION
**Pull request by oveddan**
This PR modifies the `posenet.load` function to allow on object containing `width` and `height` to be specified as input resolution, instead of just a single value.  Previously, specifying a single value always resulted in the image being resized to a square, causing a loss in accuracy.  See [#1844](https://github.com/tensorflow/tfjs/issues/1844) and [ml5 issue 510](https://github.com/ml5js/ml5-library/issues/510).

This also includes the changes from #293, which relaxes the requirement for the input resolution to be a specific value, and allows any value to be used;  internally the input resolution is adjusted to be a valid one, which is one that has its corresponding odd value divisible by the output stride.

Relaxing of the input resolution required values makes the API easier to consume and easier to understand. 

Additionally:
* Some "invalid" input resolutions were added to the demo as a form of test; internally those invalid resolutions will be adjusted to be valid resolutions.
* Added some tests around validation of the input resolution and setting it when the model is loaded.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/296)

### CODE REVIEW

**Comment by tylerzhu-github** [comment]
This is a very useful PR and I left a couple comments.
After addressing them, the PR is good to be merged

**Comment by dsmilkov** [comment]
Thanks for the quick turnaround. Want for Tyler to approve and we can merge this.

**Comment by tylerzhu-github** [comment]
After exposing the height and width in the demo. The PR is ready to be merged.
Thanks Dan! This really adds additional flexibility to our users.

**Comment by tylerzhu-github** [comment]
> After thinking about it, that user interface would be confusing to the users as they wouldn't be able to tell the difference between height and width resizing as it's not really displayed. Let's just keep it the way it is now and have another PR to review the demo ui changes. It's better to get this in and help everyone improve their accuracy and upgrade to 2.0 than hold it up due to a demo change.
> 
> > _[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/296#-:-LqYp_El0fKOX85VhPIU:bdl5z82)_ status: :shipit: complete! 1 of 1 approvals obtained (waiting on @oveddan and @tylerzhu-github)

Will merge the PR tomorrow as the first thing when i am at office.

File: posenet/src/posenet_model.ts [hunk 1]
```diff
@@ -78,6 +76,15 @@ export interface BaseModel {
  * at the cost of accuracy. Stride 32 is supported for ResNet and
  * stride 8,16,32 are supported for various MobileNetV1 models.
  *
+ * * `inputResolution`: A number or an Object of type  {width: number, height:
+ * number}. Specifies the size the input image is scaled to before feeding it
```
Comment (tylerzhu-github) [inline posenet/src/posenet_model.ts]: remove the extra space after "Ojbect of type"

File: posenet/src/posenet_test.ts [hunk 2]
```diff
@@ -37,24 +38,21 @@ describeWithFlags('PoseNet', NODE_ENVS, () => {
   beforeAll((done) => {
     // Mock out the actual load so we don't make network requests in the unit
     // test.
-    const resNetConfig = {
-      architecture: 'ResNet50',
-      outputStride: outputStride,
-      inputResolution,
-      quantBytes: quantBytes
-    } as posenetModel.ModelConfig;
+    const resNetConfig =
```
Comment (tylerzhu-github) [inline posenet/src/posenet_test.ts]: Could we reformat the config?

**Comment by oveddan** [comment]
Thanks for approving @tylerzhu-github !!

**Comment by oveddan** [comment]
@dsmilkov looks like @tylerzhu-github approved this.  can this please be merged?  thank you!

**Pull request merged**